parquet/src/main/java/org/apache/iceberg/parquet/ParquetDictionaryRowGroupFilter.java

```diff
@@ -199,7 +201,7 @@ public <T> Boolean lt(BoundReference<T> ref, Literal<T> lit) {
       int id = ref.fieldId();
 
       Boolean hasNonDictPage = isFallback.get(id);
-      if (hasNonDictPage == null || hasNonDictPage) {
+      if (hasNonDictPage == null || hasNonDictPage || isInt96Column(id)) {
```

Comment:
I'm a bit sceptical about having this check here. Can you elaborate why this is the right place to perform the check?

Reply:
I would have expected something like this:
```
--- a/parquet/src/main/java/org/apache/iceberg/parquet/ParquetDictionaryRowGroupFilter.java
+++ b/parquet/src/main/java/org/apache/iceberg/parquet/ParquetDictionaryRowGroupFilter.java
@@ -19,6 +19,7 @@
 package org.apache.iceberg.parquet;
 
 import java.io.IOException;
+import java.nio.ByteOrder;
 import java.util.Comparator;
 import java.util.Map;
 import java.util.Set;
@@ -447,6 +448,13 @@ public class ParquetDictionaryRowGroupFilter {
           case INT64:
             dictSet.add((T) conversion.apply(dict.decodeToLong(i)));
             break;
+          case INT96:
+            dictSet.add(
+                (T)
+                    conversion.apply(
+                        ParquetUtil.extractTimestampInt96(
+                            dict.decodeToBinary(i).toByteBuffer().order(ByteOrder.LITTLE_ENDIAN))));
+            break;
```

Reply:
Indeed, I've updated the patch. Please review again.